app/src/main/java/com/spascoding/taskycourse/core/data/util/RequestHelper.kt

```diff
@@ -0,0 +1,69 @@
+package com.spascoding.taskycourse.core.data.util
+
+import com.spascoding.taskycourse.core.data.Result
+import com.spascoding.taskycourse.core.domain.DataError
+import org.json.JSONObject
+import retrofit2.HttpException
+import retrofit2.Response
+import java.io.IOException
+
+object RequestHelper {
+    suspend fun <T> performRequest(
+        request: suspend () -> Response<T>,
+        onSuccess: suspend (T) -> Unit = {}
+    ): Result<T, DataError.Remote> {
+        return try {
+            val response = request()
+            if (response.isSuccessful) {
+                val body = response.body()
+                return if (body == null) {
+                    @Suppress("UNCHECKED_CAST")
+                    onSuccess(Unit as T)
```

Comment:
This cast doesn't make sense to me, how do you want to cast a Unit into whatever you passed for T? I think you just want `onSuccess(Unit)`

Reply:
![Untitled](https://github.com/spasarnaudov/TaskyCourse/assets/16733715/0eb89c46-670c-45e1-8833-e8e82a79d199)
Without casting has compile error

Reply:
So you'd want to pass the response body in the lambda to return it as a result, since you have a generic request function that should work for all cases (where data is returned and where it isn't). So doing `onSuccess(body)` should work. 
In that case, you would also need to make the generic parameter in onSuccess nullable: `onSuccess: suspend (T?) -> Unit`